src/inputs/Table/components/DeleteRowButton/DeleteRowButton.js

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useCallback, useState } from 'react';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { IconButton } from '@mui/material';
+import PropTypes from 'prop-types';
+import { FadingTooltip } from '../../../../misc';
+import { DontAskAgainDialog } from '../../../../dialogs';
+
+export const DeleteRowButton = (props) => {
+  const { onDeleteRow, labels, disabled, getSelectedRowsCount } = props;
+  const [isOpen, setOpen] = useState(false);
+  const [dialogLabel, setDialogLabel] = useState({
+    ...labels.dialog,
+    body: labels.dialog.body(getSelectedRowsCount()),
+  });
+
+  const switchDontAskAgainToDeleteRow = useCallback((isChecked) => {
+    localStorage.setItem('dontAskAgainToDeleteRow', isChecked);
+  }, []);
+
+  const confirmAndDeleteRows = useCallback(() => {
+    const selectedRowsCount = getSelectedRowsCount();
+    if (selectedRowsCount <= 1 || localStorage.getItem('dontAskAgainToDeleteRow') === 'true') onDeleteRow();
+    else {
+      setDialogLabel({ ...dialogLabel, body: labels.dialog.body(selectedRowsCount) });
+      setOpen(true);
+    }
+  }, [dialogLabel, getSelectedRowsCount, labels.dialog, onDeleteRow]);
+
+  return (
+    <>
+      <FadingTooltip title={labels.tooltip} useSpan={true}>
+        <IconButton
+          onClick={confirmAndDeleteRows}
+          component="label"
+          size="small"
+          color="primary"
+          data-cy="delete-rows-button"
+          disabled={disabled}
+          variant="text"
+        >
+          <DeleteIcon fontSize="inherit" />
+        </IconButton>
+      </FadingTooltip>
+      <DontAskAgainDialog
+        id="dontAskAgainDeleteRow"
+        onClose={() => setOpen(false)}
+        onConfirm={(isChecked) => {
+          setOpen(false);
+          switchDontAskAgainToDeleteRow(isChecked);
+          onDeleteRow();
+        }}
+        open={isOpen}
+        labels={dialogLabel}
+        confirmColor="error"
+      ></DontAskAgainDialog>
+    </>
+  );
+};
+
+DeleteRowButton.propTypes = {
+  /*
+   * Function use to delete selected lines
+   */
+  onDeleteRow: PropTypes.func.isRequired,
+  /**
+   * Component's labels:
+   * Structure:
+   * <pre>
+     {
+      tooltip: 'string',
+      dialog: {
+        title: 'string',
+        body: 'string',
+        cancel: 'string',
+        confirm: 'string',
+        checkbox: 'string',
+      },
+    }
+   </pre>
+   */
+  labels: PropTypes.shape({
+    tooltip: PropTypes.string,
+    dialog: PropTypes.shape({
+      title: PropTypes.string,
+      body: PropTypes.func,
+      cancel: PropTypes.string,
+      confirm: PropTypes.string,
+      checkbox: PropTypes.string,
+    }),
+  }),
+  /*
+   * Boolean to define if the button is disabled
+   */
+  disabled: PropTypes.bool.isRequired,
+  /*
+   * Function return the number of rows selected and update the delete row dialog body text
```

Comment:
Phrasing is not clear, do you mean _Function that returns the number of selected rows and_ **not clear** ?

Reply:
Beg your pardon?